agent/fw_slim.c

```diff
@@ -97,6 +98,59 @@ NR_PHP_WRAPPER(nr_slim3_4_route_run) {
 }
 NR_PHP_WRAPPER_END
 
+static char* nr_slim_version(zval* app TSRMLS_DC) {
+  char* retval = NULL;
+  zval* version = NULL;
+  zend_class_entry* ce = NULL;
+
+  if (0 == nr_php_is_zval_valid_object(app)) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application object is invalid",
+                     __func__);
+    return NULL;
+  }
+
+  ce = Z_OBJCE_P(app);
+  if (NULL == ce) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application has NULL class entry",
+                     __func__);
+    return NULL;
+  }
+
+  version = nr_php_get_class_constant(ce, "VERSION");
+  if (NULL == version) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application does not have VERSION",
+                     __func__);
+    return NULL;
+  }
+
+  if (nr_php_is_zval_valid_string(version)) {
+    retval = nr_strndup(Z_STRVAL_P(version), Z_STRLEN_P(version));
+  } else {
+    nrl_verbosedebug(NRL_FRAMEWORK,
+                     "%s: expected VERSION be a valid string, got type %d",
+                     __func__, Z_TYPE_P(version));
+  }
+
+  nr_php_zval_free(&version);
+  return retval;
+}
+
+NR_PHP_WRAPPER(nr_slim_application_construct) {
+  zval* this_var = nr_php_scope_get(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
```

Comment:
I believe you can remove the `TSMRLS_CC` from all the code you are adding.

Reply:
I have removed all occurences of `TSMRLS_CC`, `TSRMLS_D`, `TSRMLS_DC` in the new code I have added: [bc8b469](https://github.com/newrelic/newrelic-php-agent/pull/775/commits/bc8b4695646a148bc1ffd0a127d2d10126df8923)